qiskit/tools/qi/pauli.py

```diff
@@ -41,262 +43,499 @@ class Pauli:
+    case 'weight' is ordered by Pauli weights and
+    case 'tensor' is ordered by I,X,Y,Z counting last qubit fastest.
 
     Args:
         number_of_qubits (int): number of qubits
-        case (int): determines ordering of group elements (0=weight, 1=tensor)
+        case (str): determines ordering of group elements ('weight' or 'tensor')
 
     Returns:
         list: list of Pauli objects
 
-    Note:
-        WARNING THIS IS EXPONENTIAL
+    Raises:
+        QISKitError: case is not 'weight' or 'tensor'
+        QISKitError: number_of_qubits is larger than 4
     """
     if number_of_qubits < 5:
         temp_set = []
-        if case == 0:
-            tmp = pauli_group(number_of_qubits, case=1)
+        if case == 'weight':
+            tmp = pauli_group(number_of_qubits, case='tensor')
             # sort on the weight of the Pauli operator
             return sorted(tmp, key=lambda x: -np.count_nonzero(
                 np.array(x.to_label(), 'c') == b'I'))
-
-        elif case == 1:
+        elif case == 'tensor':
             # the Pauli set is in tensor order II IX IY IZ XI ...
-            for k_index in range(4 ** number_of_qubits):
-                v = np.zeros(number_of_qubits)
-                w = np.zeros(number_of_qubits)
+            for k in range(4 ** number_of_qubits):
+                z = np.zeros(number_of_qubits, dtype=np.bool)
+                x = np.zeros(number_of_qubits, dtype=np.bool)
                 # looping over all the qubits
-                for j_index in range(number_of_qubits):
-                    # making the Pauli for each kindex i fill it in from the
+                for j in range(number_of_qubits):
+                    # making the Pauli for each j fill it in from the
                     # end first
-                    element = int((k_index) / (4 ** (j_index))) % 4
-                    if element == 0:
-                        v[j_index] = 0
-                        w[j_index] = 0
-                    elif element == 1:
-                        v[j_index] = 0
-                        w[j_index] = 1
+                    element = (k // (4 ** j)) % 4
+                    if element == 1:
+                        x[j] = True
                     elif element == 2:
-                        v[j_index] = 1
-                        w[j_index] = 1
+                        z[j] = True
+                        x[j] = True
                     elif element == 3:
-                        v[j_index] = 1
-                        w[j_index] = 0
-                temp_set.append(Pauli(v, w))
+                        z[j] = True
+                temp_set.append(Pauli(z, x))
             return temp_set
+        else:
+            raise QISKitError("Only support 'weight' or 'tensor' cases")
 
-    print('please set the number of qubits to less than 5')
-    return -1
+    raise QISKitError("Only support number of qubits is less than 5")
 
 
 def pauli_singles(j_index, number_qubits):
```

Comment:
looking at the code this returns the three Pauli single qubit operators x y z for qubit i. I don't know why we need this. Does anyone

Reply:
removed. and having a classmethod to create a single pauli easily